Material_Button/src/main/ets/components/ToggleButtonModel.ets

```diff
@@ -0,0 +1,136 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ToggleButtonOptions, ToggleSelection, ButtonVariant, ButtonOptions, TextualToggleType, IconToggleType } from "./Types"
+import { ButtonModel } from "./ButtonModel"
+
+export class ToggleButtonModel implements ToggleButtonOptions{
+  data: TextualToggleType[] | IconToggleType[] = []
+  width: number = 40
+  height: number = 40
+  selectionType: ToggleSelection = ToggleSelection.Single
+  toggleBorderWidth: number = 2
+  selectedBorderWidth: number = 2
+  backgroundColor: ResourceColor = Color.White
+  toggleBorderColor: ResourceColor = Color.Grey
+  toggleBorderRadius: number = undefined
+  selectedBackgroundColor: ResourceColor = "#D3D3D3"
+  fontColor: ResourceColor = Color.White
+  fontSize: number = 20
+  checkedVariant: ButtonVariant = ButtonVariant.TEXT
+  checkedBackgroundColor: ResourceColor = undefined
+  toggleButtonModel: ToggleButtonModel = undefined
+  buttonModelList: ButtonModel[] = []
+  buttonSelStatus: boolean[] = []
+
+  constructor(inputOptions: ToggleButtonOptions) {
+    this.update(inputOptions)
+    let buttonOptions: ButtonOptions = undefined
+    this.data.forEach((entry, index) => {
+      buttonOptions = undefined
+      if ('content' in entry) {
+        let opacity = 1
+        if (typeof entry.enable == 'boolean' && !entry.enable) {
+          opacity = 0.38
+        }
+        buttonOptions = {
+          variant: ButtonVariant.TEXT,
+          content: entry.content,
+          responseCode: index,
+          backgroundColor: this.backgroundColor,
+          height: this.height,
+          width: this.width,
+          opacity: opacity,
+          ripple:false
+        }
+      } else {
+        let opacity = 1
+        if (typeof entry.enable == 'boolean' && !entry.enable) {
+          opacity = 0.38
+        }
+        buttonOptions = {
+          variant: ButtonVariant.TEXT,
+          iconSrc: entry.iconSrc,
+          responseCode: index,
+          height: this.height,
+          width: this.width,
+          backgroundColor: this.backgroundColor,
+          opacity:opacity,
+          ripple:false
+        }
+      }
+      if (index > 0) {
+        buttonOptions['margin'] = {left: this.toggleBorderWidth}
+      }
+      let bm = new ButtonModel(buttonOptions)
+      bm.update(inputOptions)
+      this.buttonModelList.push(bm)
+      this.buttonSelStatus.push(false)
+      if (entry.checked) {
+        this.checkOrUncheck(index)
+      }
+    })
+  }
+
+  update(options: ToggleButtonOptions) {
+    if (options) {
+      if (options.data != undefined) this.data = options.data
```

Comment:
better to add null check also.

Reply:
var x = null //Or unknown

if (x) {//fails for both null and unknown
    console.log("inside")
}
console.log("outside")

as this code handles both, we need not add explicitly.